packages/core/src/utils/signature.ts

```diff
@@ -0,0 +1,10 @@
+import { createHmac } from 'node:crypto';
+
+import { canonicalize } from 'json-canonicalize';
+
+export const generateSignature = (signingKey: string, payload: Record<string, unknown>) => {
+  const hmac = createHmac('sha256', signingKey);
+  const payloadString = canonicalize(payload);
+  hmac.update(payloadString);
+  return `sha256=${hmac.digest('hex')}`;
+};
```

Comment:
maybe no need for the new dependency? https://github.com/panva/jose/blob/main/src/runtime/node/sign.ts

Reply:
https://github.com/panva/jose/blob/main/src/jws/flattened/sign.ts#LL120C7-L120C51

Reply:
just use `TextEncoder` and base64

Reply:
Per offline discussion, we just need to stringify the payload data.
Updated.